snapcraft/parts/lifecycle.py

```diff
@@ -87,16 +93,19 @@ def process_yaml(project_file: Path) -> Dict[str, Any]:
             msg = f"{msg}: {err.filename!r}."
         raise errors.SnapcraftError(msg) from err
 
-    # validate project grammar
+    return yaml_data
+
+
+def process_yaml(yaml_data, build_to: str) -> Dict[str, Any]:
+    """Process the yaml from project file."""
     GrammarAwareProject.validate_grammar(yaml_data)
 
-    # TODO: support for target_arch
     arch = _get_arch()
-    yaml_data = extensions.apply_extensions(yaml_data, arch=arch, target_arch=arch)
+    yaml_data = extensions.apply_extensions(yaml_data, arch=arch, target_arch=build_to)
```

Comment:
build_to is not assigned anywhere yet, right?

Reply:
I'm assigning an architecture to `build_to` in `get_build_plan()`

Reply:
sorry, I am just seeing the param in process_yaml now